Support for React 17.x

Could give support for React 17.x. I'm using with react 17.x but must install deps with --force.

Yes, it's because of peerDeps.

```bash
npm ERR! While resolving: react-fuzzy-highlighter@1.0.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"17.0.2" from the root project
npm ERR!   peer react@"^16 || ^17 || ^18" from @headlessui/react@1.3.0
npm ERR!   node_modules/@headlessui/react
npm ERR!     @headlessui/react@"1.3.0" from the root project
npm ERR!   16 more (@heroicons/react, @next/react-dev-overlay, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.x" from react-fuzzy-highlighter@1.0.0
npm ERR! node_modules/react-fuzzy-highlighter
npm ERR!   react-fuzzy-highlighter@"1.0.0" from the root project
```

--- a/package.json
+++ b/package.json
@@ -18,10 +18,6 @@
     "set-value": "^3.0.1",
     "strind": "^0.3.0"
   },
-  "peerDependencies": {
-    "react": "^16.x",
-    "react-dom": "^16.x"
-  },
   "devDependencies": {
     "@types/enzyme": "^3.10.3",
     "@types/enzyme-adapter-react-16": "^1.0.5",

